compiler/rustc_data_structures/src/steal.rs

```diff
@@ -30,17 +30,19 @@ impl<T> Steal<T> {
         Steal { value: RwLock::new(Some(value)) }
     }
 
+    #[track_caller]
     pub fn borrow(&self) -> MappedReadGuard<'_, T> {
         ReadGuard::map(self.value.borrow(), |opt| match *opt {
             None => panic!("attempted to read from stolen value"),
```

Comment:
This PR only fixed `.steal()`, `.borrow()` has no effect because the `panic!` is in a closure.